app/controllers/sms_controller.rb

```diff
@@ -73,11 +73,16 @@ def incoming_msg
   def deliver_reply(reply)
     if reply
       if incoming_adapter.reply_style == :via_adapter
-        outgoing_adapter.deliver(reply)
+        begin
+          raise Sms::Error, ENV["STUB_REPLY_ERROR"] if Rails.env.test? && ENV["STUB_REPLY_ERROR"].present?
+          outgoing_adapter.deliver(reply)
+        rescue Sms::Error => e
+          reply.error_message = e
+          reply.save
```

Comment:
Remember: never call `save` by itself unless you're prepared to deal with validation errors. Use `save!` if there shouldn't be any or `save(validate: false)` if you want to skip validation. `save!` should work here.